crates/aiken-lang/src/builtins.rs

```diff
@@ -320,6 +321,77 @@ pub fn prelude(id_gen: &IdGenerator) -> TypeInfo {
         },
     );
 
+    // Pair(a, b)
+    let fst_parameter = generic_var(id_gen.next());
+    let snd_parameter = generic_var(id_gen.next());
+    prelude.types.insert(
+        PAIR.to_string(),
+        TypeConstructor {
+            location: Span::empty(),
+            parameters: vec![fst_parameter.clone(), snd_parameter.clone()],
+            tipo: pair(fst_parameter.clone(), snd_parameter.clone()),
+            module: "".to_string(),
+            public: true,
+        },
+    );
+
+    prelude
+        .types_constructors
+        .insert(PAIR.to_string(), vec![PAIR.to_string()]);
+
+    let mut pair_fields = HashMap::new();
+    pair_fields.insert("fst".to_string(), (0, Span::empty()));
+    pair_fields.insert("snd".to_string(), (1, Span::empty()));
+    prelude.values.insert(
+        PAIR.to_string(),
+        ValueConstructor::public(
+            function(
+                vec![fst_parameter.clone(), snd_parameter.clone()],
+                pair(fst_parameter.clone(), snd_parameter.clone()),
+            ),
+            ValueConstructorVariant::Record {
+                module: "".into(),
+                name: PAIR.to_string(),
+                field_map: Some(FieldMap {
+                    arity: 2,
+                    fields: pair_fields,
+                    is_function: false,
+                }),
+                arity: 2,
+                location: Span::empty(),
+                constructors_count: 1,
+            },
+        ),
+    );
```

Comment:
I don't think this will work, unless we introduce some special magic in both codegen and the type checker for it. Rather, we should leverage the tuple-access syntax _at the parser level_ and interpret those as calls to builtins.

Reply:
It does work, I changed code gen.